plugin/pkg/scheduler/algorithmprovider/defaults/defaults.go

```diff
@@ -80,6 +89,27 @@ func defaultPredicates() sets.String {
 		),
 		// Fit is determined by the presence of the Host parameter and a string match
 		factory.RegisterFitPredicate("HostName", predicates.PodFitsHost),
+
+		// Fit is determined by whether or not there would be too many AWS EBS volumes attached to the node
+		factory.RegisterFitPredicateFactory(
+			"MaxEBSVolumeCount",
+			func(args factory.PluginFactoryArgs) algorithm.FitPredicate {
+				// According to Amazon, attaching more than 40 volumes can cause boot failures
+				// TODO: allow for generically parameterized scheduler predicates, because this is a bit ugly
+				var maxEBSVols uint = DefaultMaxEBSVolumes
+				if rawMaxEBSVols := os.Getenv("KUBE_MAX_EBS_VOLS"); rawMaxEBSVols != "" {
```

Comment:
Is there a legit reason why you would over-ride this value?  Also the env-var seems pretty hacky.